slack/resource_conversation.go

```diff
@@ -107,6 +114,17 @@ func resourceSlackConversationCreate(ctx context.Context, d *schema.ResourceData
 	isPrivate := d.Get("is_private").(bool)
 
 	channel, err := client.CreateConversationContext(ctx, name, isPrivate)
+	if err != nil && err.Error() == "name_taken" && d.Get("adopt_existing_channel").(bool) {
+		channel, err = findExistingChannel(ctx, client, name, isPrivate)
+		if err == nil && channel.IsArchived {
+			// ensure unarchived first if adopting existing channel, else other calls below will fail
+			if err := client.UnArchiveConversationContext(ctx, channel.ID); err != nil {
+				if err.Error() != "not_archived" {
+					return diag.Errorf("couldn't unarchive conversation %s: %s", channel.ID, err)
```

Comment:
we really should never get this error, as we check that it's archived before unarchived, but I guess there could be race conditions with a user unarchiving it manually in between the two calls.